searchcore/src/vespa/searchcore/proton/matching/matcher.cpp

```diff
@@ -286,19 +291,20 @@ Matcher::match(const SearchRequest &request, vespalib::ThreadBundle &threadBundl
         _stats.add(my_stats);
         if (my_stats.softDoomed()) {
             double old = _stats.softDoomFactor();
-            fastos::TimeStamp softLimit = (1.0 - _rankSetup->getSoftTimeoutTailCost()) * request.getTimeout();
+            fastos::TimeStamp overtimeLimit = (1.0 - _rankSetup->getSoftTimeoutTailCost()) * request.getTimeout();
             fastos::TimeStamp adjustedDuration = duration - my_stats.doomOvertime();
             if (adjustedDuration < 0) {
                 adjustedDuration = 0;
             }
             bool allowedSoftTimeoutFactorAdjustment = (std::chrono::duration_cast<std::chrono::seconds>(my_clock::now() - _startTime).count() > SECONDS_BEFORE_ALLOWING_SOFT_TIMEOUT_FACTOR_ADJUSTMENT);
-            if (allowedSoftTimeoutFactorAdjustment) {
-                _stats.updatesoftDoomFactor(request.getTimeout(), softLimit, adjustedDuration);
+            if (allowedSoftTimeoutFactorAdjustment && ! isDoomExplicit) {
```

Comment:
consider making the explicit doom part of whether we are allowed to tamper, not an additional reason not to

Reply:
Done